sfepy/discrete/common/fields.py

```diff
@@ -453,7 +452,31 @@ def evaluate_at(self, coors, source_vals, mode='val', strategy='general',
         output('...done',verbose=verbose)
 
         if mode == 'val':
-            vals.shape = (coors.shape[0], source_vals.shape[1])
+            vals.shape = (nc, n_comp)
+
+        elif mode == 'div':
+            assert_(n_comp == dim)
+            vals = nm.trace(vals, axis1=1, axis2=2).reshape(nc, 1, 1)
+
+        elif mode == 'cauchy_strain':
+            assert_(n_comp == dim)
+            sym = (dim + 1) * dim // 2
+            aux = nm.empty((nc, sym, 1), dtype=source_dtype)
+
+            strain_tab = {
+                1: (0, ),
+                2: (0, 3, (1, 2)),
+                3: (0, 4, 8, (1, 3), (2, 6), (5, 8)),
+            }
+
+            vals_ = vals.reshape((nc, -1))
+            for ii, idx in enumerate(strain_tab[dim]):
+                if isinstance(idx, tuple):
+                    aux[:, ii, 0] = nm.sum(vals_[:, nm.array(idx)], axis=1)
+                else:
+                    aux[:, ii, 0] = vals_[:, idx]
+
+            vals = aux
```

Comment:
This could be made into a function (get_symgrad() or get_cauchy_strain()?) and put into sfepy.mechanics.tensors among the other get_* functions, what do you think?

Reply:
I moved the code into `sfepy.mechanics.tensors.get_cauchy_strain()`.